app/src/main/java/com/internship/move/presentation/map/MapFragment.kt

```diff
@@ -1,35 +1,114 @@
 package com.internship.move.presentation.map
 
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.pm.PackageManager
+import android.location.Location
 import android.os.Bundle
 import android.view.View
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
 import androidx.fragment.app.Fragment
 import androidx.navigation.fragment.findNavController
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MapStyleOptions
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.gms.tasks.Task
 import com.internship.move.R
 import com.internship.move.databinding.FragmentMapBinding
 import com.internship.move.presentation.map.viewmodel.MapViewModel
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class MapFragment : Fragment(R.layout.fragment_map) {
+class MapFragment : Fragment(R.layout.fragment_map), OnMapReadyCallback {
 
     private val binding by viewBinding(FragmentMapBinding::bind)
     private val viewModel by viewModel<MapViewModel>()
+    private lateinit var map: GoogleMap
```

Comment:
I believe map and fusedLocationProviderClient should be nullable and that you should clear them at onDestroyView in order to avoid memory leaks
```
map?.clear()
map = null
fusedLocationProviderClient = null
```